### PR TITLE
Add filebeat recipes

### PR DESCRIPTION
**Pull request by gmacdougall**
This adds filebeat for shipping logs off to elastic search.

### CODE REVIEW

File: filebeat/recipes/default.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+template '/etc/yum.repos.d/filebeat.repo' do
+  source 'filebeat.repo.erb'
+  owner 'root'
+  group 'root'
+  mode '0755'
+end
+
+execute 'sudo rpm --import https://packages.elastic.co/GPG-KEY-elasticsearch'
+execute 'sudo yum install -y filebeat'
```
Comment (stewart) [inline filebeat/recipes/default.rb]: Should this be done with `yum_package` instead?

File: filebeat/recipes/default.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+template '/etc/yum.repos.d/filebeat.repo' do
+  source 'filebeat.repo.erb'
+  owner 'root'
+  group 'root'
+  mode '0755'
+end
```
Comment (stewart) [inline filebeat/recipes/default.rb]: Should this be done with `yum_repository` instead?

Comment (gmacdougall) [inline filebeat/recipes/default.rb]: As far as I know `yum_repository` is Chef 12 only.

Comment (stewart) [inline filebeat/recipes/default.rb]: It looks to be provided by the [version of `yum` in the repo](https://github.com/chef-cookbooks/yum/tree/4e53fe10d713fe04219c81bb4a743eccba34c6a5).

File: filebeat/recipes/default.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+execute 'sudo rpm --import https://packages.elastic.co/GPG-KEY-elasticsearch'
+
+yum_package 'filbeat'
```
Comment (stewart) [inline filebeat/recipes/default.rb]: Typo.

File: filebeat/README.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+Filebeat Opsworks recipe
+========================
+
+Chef cookbook to install and configure filebeat service to send logs to remote
+elasticache server
+
+## Restrictions
+
+This currently only works with Amazon Linux or other yum based ditributions
```
Comment (stewart) [inline filebeat/README.md]: Typo.

File: filebeat/templates/default/filebeat.repo.erb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+[elastic-5.x]
```
Comment (stewart) [inline filebeat/templates/default/filebeat.repo.erb]: This file should be able to be removed now.

File: filebeat/recipes/default.rb (hunk 2 above)
Comment (gmacdougall) [inline filebeat/recipes/default.rb]: I do get
```
NoMethodError
-------------
No resource or method named `yum_repository' for `Chef::Recipe "default"'
 ```

when attempting to use `yum_repository`

File: filebeat/recipes/default.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
+template '/etc/yum.repos.d/filebeat.repo' do
+  source 'filebeat.repo.erb'
+  owner 'root'
+  group 'root'
+  mode '0755'
```
Comment (alexblackie) [inline filebeat/recipes/default.rb]: probably don't need exec on this

File: filebeat/recipes/default.rb (hunk 2 above)
Comment (alexblackie) [inline filebeat/recipes/default.rb]: You may need to `depends "yum"` in `metadata.rb` in order to get things from it.

File: filebeat/recipes/default.rb [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+execute 'sudo rpm --import https://packages.elastic.co/GPG-KEY-elasticsearch'
+
+yum_package 'filebeat'
+
+template '/etc/filebeat/filebeat.yml' do
+  source 'filebeat.yml.erb'
+  owner 'root'
+  group 'root'
+  mode '0755'
```
Comment (alexblackie) [inline filebeat/recipes/default.rb]: Again, probably don't need exec on this file.

File: filebeat/templates/default/filebeat.yml.erb [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+  paths:
+  <% @paths.each do |path| %>
+    <%= "- #{path}" %>
+  <% end %>
+
+  json.message_key: event
+  json.keys_under_root: true
+
+output.elasticsearch:
+  hosts: <%= @hosts %>
```
Comment (alexblackie) [inline filebeat/templates/default/filebeat.yml.erb]: should this be nested as a loop (as above with `paths`)? unclear from just reading it if this stringifies to valid yaml with multiple values

**Comment by gmacdougall** [comment]
`depends 'yum'` was the thing that I was missing. Thanks @alexblackie!

**Pull request merged**